### PR TITLE
Conditionally run BigQuery tests which require GCE access

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -275,6 +276,28 @@ jobs:
...
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Maven Install
+        run: |
+          export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
+          $RETRY ./mvnw install ${MAVEN_FAST_INSTALL} -am -pl :trino-memsql
```
Comment (ebyhr) [inline .github/workflows/ci.yml:292]:
```suggestion
          $RETRY ./mvnw install ${MAVEN_FAST_INSTALL} -am -pl :trino-bigquery
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -275,6 +276,28 @@ jobs:
             ./mvnw test ${MAVEN_TEST} -pl :trino-memsql -Dmemsql.license=${MEMSQL_LICENSE}
           fi
 
+  test-bigquery:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
```
Comment (ebyhr) [inline .github/workflows/ci.yml:282]: Though I guess it comes from `test-memsql`, `strategy.fail-fast` is for matrix build in my understanding.

Comment (losipiuk) [reply .github/workflows/ci.yml:282]: Yeah - will drop.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (losipiuk) [reply .github/workflows/ci.yml]: Thx

**Comment by losipiuk** [comment]
@ebyhr I reworked it a bit so the tests which do not require GCE access are always run. PTAL

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -218,7 +218,8 @@ jobs:
             !:trino-kudu,
             !:trino-phoenix,!:trino-iceberg,!:trino-druid,
             !:trino-docs,!:trino-server,!:trino-server-rpm,
-            !:trino-test-jdbc-compatibility-old-server'
+            !:trino-test-jdbc-compatibility-old-server,
+            !:trino-bigquery'
```
Comment (findepi) [inline .github/workflows/ci.yml:222]: the exclusions here should be in the same order as the tests definitions are, but i see memsql is already out of place.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -288,8 +288,15 @@ jobs:
...
           $RETRY ./mvnw install ${MAVEN_FAST_INSTALL} -am -pl :trino-bigquery
-      - name: BigQuery Tests
+      - name: Basic BigQuery Tests
         run: ./mvnw test ${MAVEN_TEST} -pl :trino-bigquery
+      - name: Cloud BigQuery Tests
+        env:
+          BIGQUERY_CREDENTIALS_KEY: ${{ secrets.BIGQUERY_CREDENTIALS_KEY }}
+        run: |
+          if [ "${BIGQUERY_CREDENTIALS_KEY}" != "" ]; then
+            ./mvnw test ${MAVEN_TEST} -pl :trino-bigquery -Pcloud-tests -Dbigquery.credentials-key=${BIGQUERY_CREDENTIALS_KEY}
```
Comment (findepi) [inline .github/workflows/ci.yml:296]: put `${BIGQUERY_CREDENTIALS_KEY}` in `"` quotes

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -256,8 +256,6 @@ jobs:
 
   test-memsql:
     runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
```
Comment (findepi) [inline .github/workflows/ci.yml:259]: done by @ebyhr in some other PR

**Review by findepi — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (losipiuk) [reply .github/workflows/ci.yml:222]: memsql and bigquery are separate jobs. So I am not sure what you suggest here.

File: .github/workflows/ci.yml (hunk 5 above)
Comment (losipiuk) [reply .github/workflows/ci.yml]: merged that one :)

**Review by ebyhr — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (findepi) [reply .github/workflows/ci.yml:222]: order the exclusions (the `!...` lines) here in order of test appearance elsewhere -- the `test` matrix, and dedicated test jobs (memsql, bq).

**Pull request merged**